src/adapters/net.cpp

```diff
@@ -28,6 +30,38 @@ namespace net {
     return file_util::exists("/sys/class/net/" + ifname + "/wireless");
   }
 
+  std::string find_wireless_interface() {
+    DIR* dir = opendir("/sys/class/net");
+    while (true) {
+      struct dirent* entry = readdir(dir);
```

Comment:
From `man readdir`:

> readdir() is not required to be thread-safe.  However, in modern implementations (including the glibc implementation), concurrent calls to readdir() that specify different directory streams are thread-safe.

If there are multiple network modules, we may run into race conditions here since they all will open the same directory. We will probably not get around using a mutex here.